intTests/runtests.sh

```diff
@@ -85,63 +105,69 @@ XML_TEMP="${XML_FILE}.tmp"
 rm -f logs/*
 
-NUM_TESTS=0
+NUM_TESTS=$(echo "$TESTS" | wc -w)
+PASSED_TESTS=0
 FAILED_TESTS=0
 FAILED_TEST_DETAILS="failed:"
 export TIMEFORMAT="%R"
 TOTAL_TIME=0
 
 for i in $TESTS; do
-  NUM_TESTS=$(( $NUM_TESTS + 1 ))
-
   # Some nasty bash hacking here to catpure the amount of time taken by the test
   # See http://mywiki.wooledge.org/BashFAQ/032
   START_TIME=$SECONDS
   if [ "${OS}" == "Windows_NT" ]; then
     # ulimit is useless on cygwin :-(  Use the 'timeout' utility instead
-    (cd $i; (/usr/bin/timeout -k 15 ${TEST_TIMEOUT} sh -vx test.sh > ../logs/$i.log 2>&1) 2>&1 )
+    ( cd "$i" || exit 1; (/usr/bin/timeout -k 15 ${TEST_TIMEOUT} sh -vx test.sh > "../logs/$i.log" 2>&1) 2>&1 )
     RES=$?
   else
-    (ulimit -t ${TEST_TIMEOUT}; cd $i; (sh -vx test.sh > ../logs/$i.log 2>&1) 2>&1 )
+    ( ulimit -t ${TEST_TIMEOUT}; cd "$i" || exit 1; (sh -vx test.sh > "../logs/$i.log" 2>&1) 2>&1 )
     RES=$?
   fi
   END_TIME=$SECONDS
-  TEST_TIME=$(expr $END_TIME - $START_TIME)
+  TEST_TIME=$((END_TIME - START_TIME))
 
-  if [ $(expr $TEST_TIME ">=" $TEST_TIMEOUT) == 1 ]; then
+  if [ $((TEST_TIME >= TEST_TIMEOUT)) == 1 ]; then
     TIMED_OUT=" TIMEOUT"
   else
     TIMED_OUT=""
   fi
 
-  TOTAL_TIME=$(expr ${TOTAL_TIME} + ${TEST_TIME})
+  TOTAL_TIME=$((TOTAL_TIME + TEST_TIME))
 
   if [ $RES == 0 ]; then
+    PASSED_TESTS=$(( PASSED_TESTS + 1 ))
     echo "$i: OK (${TEST_TIME}s)"
     echo "  <testcase name=\"${i}\" time=\"${TEST_TIME}\" />" >> ${XML_TEMP}
   else
+    FAILED_TESTS=$(( FAILED_TESTS + 1 ))
     echo "$i: FAIL (${TEST_TIME}s${TIMED_OUT})"
-    if [ ! -z "$LOUD" ]; then cat logs/$i.log; fi
-    FAILED_TESTS=$(( $FAILED_TESTS + 1 ))
+    if [ -n "$LOUD" ]; then cat "logs/$i.log"; fi
     FAILED_TEST_DETAILS+=" $i"
-    echo "  <testcase name=\"${i}\" time=\"${TEST_TIME}\"><failure><![CDATA[" >> ${XML_TEMP}
-    sed -e 's/]]>/] ]>/' logs/$i.log >> ${XML_TEMP}
-    echo "]]></failure></testcase>" >> ${XML_TEMP}
+    {
+      echo "  <testcase name=\"${i}\" time=\"${TEST_TIME}\"><failure><![CDATA["
+      sed -e 's/]]>/] ]>/' "logs/$i.log"
+      echo "]]></failure></testcase>"
+    } >> ${XML_TEMP}
   fi
 done
 
+DONE=1
+
 echo "<?xml version='1.0'?>" > $XML_FILE
-echo "<testsuites errors=\"${FAILED_TESTS}\" tests=\"${NUM_TESTS}\" time=\"${TOTAL_TIME}\">" >> $XML_FILE
-echo " <testsuite name=\"SAWScript Integration Tests\">" >> $XML_FILE
-cat $XML_TEMP >> $XML_FILE
-echo " </testsuite>" >> $XML_FILE
-echo "</testsuites>" >> $XML_FILE
+{
+  echo "<testsuites errors=\"${FAILED_TESTS}\" tests=\"${NUM_TESTS}\" time=\"${TOTAL_TIME}\">"
+  echo " <testsuite name=\"SAWScript Integration Tests\">"
+  cat $XML_TEMP
+  echo " </testsuite>"
+  echo "</testsuites>"
+} >> $XML_FILE
+
 rm $XML_TEMP
 
-PASSED_TESTS=$(expr $NUM_TESTS - $FAILED_TESTS)
-echo "tests passed: ${PASSED_TESTS} / ${NUM_TESTS}"
-if [ "${FAILED_TESTS}" == 0 ]; then
-    echo "all tests passed"
-else
-    echo "${FAILED_TEST_DETAILS}"
-fi
+finish;
+
+# Local Variables:
```

Comment:
I prefer that we avoid editor-specific content in any files, it's a slippery slope.

Reply:
Oops, threw this in to conform with existing indentation and then forgot about it.